src/onboarding/welcome/Welcome.test.tsx

```diff
@@ -1,74 +1,292 @@
-import { fireEvent, render } from '@testing-library/react-native'
+import { fireEvent, render, waitFor } from '@testing-library/react-native'
 import * as React from 'react'
 import { Provider } from 'react-redux'
+import { acceptTerms, chooseCreateAccount, chooseRestoreAccount } from 'src/account/actions'
+import { OnboardingEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
 import { firstOnboardingScreen } from 'src/onboarding/steps'
 import Welcome from 'src/onboarding/welcome/Welcome'
-import { patchUpdateStatsigUser } from 'src/statsig'
+import { getExperimentParams, patchUpdateStatsigUser } from 'src/statsig'
 import { createMockStore } from 'test/utils'
 
 jest.mock('src/onboarding/steps')
 jest.mock('src/statsig', () => ({
-  ...(jest.requireActual('src/statsig') as any),
+  getExperimentParams: jest.fn(),
   patchUpdateStatsigUser: jest.fn(),
 }))
 
 describe('Welcome', () => {
   beforeAll(() => {
     jest.spyOn(Date, 'now').mockImplementation(() => 123)
+    jest.mocked(firstOnboardingScreen).mockReturnValue(Screens.PincodeSet)
   })
-  it('renders and behaves correctly', async () => {
-    const store = createMockStore()
-    const { getByTestId } = render(
-      <Provider store={store}>
-        <Welcome /*{...getMockStackScreenProps(Screens.)}*/ />
-      </Provider>
-    )
-    fireEvent.press(getByTestId('CreateAccountButton'))
-    jest.runOnlyPendingTimers()
-    await Promise.resolve() // waits for patchUpdateStatsigUser promise to resolve
-    expect(patchUpdateStatsigUser).toHaveBeenCalledWith({ custom: { startOnboardingTime: 123 } })
-    expect(store.getActions()).toMatchInlineSnapshot(`
-      [
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe.each([{ variant: 'control' }, { variant: 'colloquial_terms' }])(
```

Comment:
nice test format to make it very straightforward to clean up the experiment when the time comes